weatherAplication/Modules/Home/View/HomeViewController.swift

```diff
@@ -2,89 +2,78 @@ import SnapKit
 import UIKit
 
 class HomeViewController: UIViewController {
-    private let kLogoWidth: CGFloat = 250
-    private let kLogoHeight: CGFloat = 250
-    private let kSideMargin: CGFloat = 10
-    private let kTopMargin: CGFloat = 20
-    private let kBottomMargin: CGFloat = 15
-    private let aplicationImage = R.image.weather_symbol()
-    private let logoImageView = UIImageView()
-    private let aplicationNameLabel = UILabel()
-    private let proceedButton = UIButton()
-    private let aplicationNameString: String = "MyWeather"
-    private let buttonString: String = R.string.localizable.button_string()
-
     private var viewModel: HomeViewModelContract
+    private var homeView = HomeView()
 
     init() {
         viewModel = HomeViewModel()
         super.init(nibName: nil, bundle: nil)
+        homeView.lastCityNamesTableView.dataSource = self
+        homeView.lastCityNamesTableView.delegate = self
+    }
+
+    override func loadView() {
+       view = homeView
     }
 
     required init?(coder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        viewModel.viewWillAppear()
+    }
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        setupView()
-        setupConstraints()
         viewModel.delegate = self
-        proceedButton.addTarget(self, action: #selector(buttonPressed), for: .touchUpInside)
+        homeView.searchButton.addTarget(self, action: #selector(buttonPressed), for: .touchUpInside)
     }
 
     @objc
     func buttonPressed() {
         viewModel.proceedButtonPressed()
     }
 
-    func setupView() {
-        view.backgroundColor = .blue
-
-        logoImageView.image = aplicationImage
-        view.addSubview(logoImageView)
-
-        aplicationNameLabel.text = aplicationNameString
-        aplicationNameLabel.textColor = .white
-        aplicationNameLabel.textAlignment = .center
-        aplicationNameLabel.backgroundColor = .black
-        view.addSubview(aplicationNameLabel)
+}
 
-        proceedButton.setTitle(buttonString, for: .normal)
-        proceedButton.setTitleColor(.red, for: .normal)
-        proceedButton.backgroundColor = .yellow
-        view.addSubview(proceedButton)
+extension HomeViewController: HomeViewModelDelegate {
+    func openSearchViewController() {
+        let nextViewController = SearchViewController()
```

Comment:
nazwa `nextViewController` nie bardzo mówi czym to jest

Reply:
☝️